tests/kubernetes-distros/kind/lumigooperator_traces_test.go

```diff
@@ -421,6 +422,49 @@ func TestLumigoOperatorTraces(t *testing.T) {
 
 			return ctx
 		}).
+		Assess("All traces have the 'k8s.provider.id' taken from describe node", func(ctx context.Context, t *testing.T, c *envconf.Config) context.Context {
+
+			otlpSinkDataPath := ctx.Value(internal.ContextKeyOtlpSinkDataPath).(string)
+
+			tracesPath := filepath.Join(otlpSinkDataPath, "traces.json")
+
+			traceBytes, err := os.ReadFile(tracesPath)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			if len(traceBytes) < 1 {
+				t.Fatalf("No trace data found in '%s'", tracesPath)
+			}
+
+			scanner := bufio.NewScanner(bytes.NewBuffer(traceBytes))
+			for scanner.Scan() {
+				exportRequest := ptraceotlp.NewExportRequest()
+				exportRequest.UnmarshalJSON(scanner.Bytes())
+
+				if exportRequest.Traces().SpanCount() < 1 {
+					continue
+				}
+
+				l := exportRequest.Traces().ResourceSpans().Len()
+				for i := 0; i < l; i++ {
+					resourceSpans := exportRequest.Traces().ResourceSpans().At(i)
+
+					resourceAttributes := resourceSpans.Resource().Attributes().AsRaw()
+
+					if actualClusterUID, found := resourceAttributes["k8s.provider.id"]; !found {
+						t.Fatalf("found spans without the 'k8s.provider.id' resource attribute: %+v", resourceAttributes)
+					} else {
+						actualClusterUIDstr, ok := actualClusterUID.(string)
+						if !ok || !strings.HasPrefix(actualClusterUIDstr, "kind://") {
```

Comment:
This test is not strong enough IMO. You actually could add the cluster name as a context param and read it in the test